src/hooks/useOnboard.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { Wallet } from "bnc-onboard/dist/src/interfaces";
+import Onboard from "bnc-onboard";
+import { ethers } from "ethers";
+
+import { useConnection } from "./useConnection";
+import { onboardBaseConfig } from "../config";
+import { UnsupportedChainIdError, isValidChainId } from "../utils/chainId";
+
+export function useOnboard() {
+  const { connect, disconnect, update, setError } = useConnection();
+  const instance = React.useMemo(
+    () =>
+      Onboard({
+        ...onboardBaseConfig(),
+        subscriptions: {
+          address: (address: string) => {
+            update({ account: address });
+          },
+          network: (networkId: number) => {
+            const error = isValidChainId(networkId)
+              ? undefined
+              : new UnsupportedChainIdError(networkId);
+            update({
+              chainId: networkId,
+            });
+            if (error) {
+              setError(error);
+            }
+          },
+          wallet: async (wallet: Wallet) => {
+            if (wallet.provider) {
+              const provider = new ethers.providers.Web3Provider(
```

Comment:
maybe consider an adapter for ethers as well 😄

Reply:
is this really the only place where you can get the provider?

Reply:
(Answering the ethers comment above) You're right that its awkward to depend on ethers, ideally you should just pass the provider and signer in and `useConnection` should take care of that. I'll think about how to make this  library agnostic! 

Regarding the second comment, I think you could call `onboard.getState()` to get it, but setting up the subscription here would be necessary regardless

Reply:
> (Answering the ethers comment above) You're right that its awkward to depend on ethers, ideally you should just pass the provider and signer in and useConnection should take care of that. I'll think about how to make this library agnostic!

yea i dont think we need to go crazy with this, maybe it means just moving where you wrap the provider in ethers, might be moved to the connection hook itself, or a useEthers hook or something like that. 

> Regarding the second comment, I think you could call onboard.getState() to get it, but setting up the subscription here would be necessary regardless

given that you can just dispatch the provider at any point, i think its fine as is, was just curious, as it seems there should be a onProvider callback, or something like that.

Reply:
Moved it over to `useConnection` , now `useOnboard` has no dep on ethers